test/wallet-rpc-test.js

```diff
@@ -0,0 +1,315 @@
+    });
+  }
+
+  /**
+   * Close util and all its child objects.
+   */
+
+  async close() {
+    assert(this.opened, 'TestUtil is not open.');
+    this.opened = false;
+
+    await this.nclient.close();
+    await this.wclient.close();
+    await this.node.close();
+  }
+
+  async confirmTX(txid, timeout = 5000) {
+    return common.forValue(this.txs, txid, txid, timeout);
+  }
+
+  async confirmBlock(hash, timeout = 5000) {
+    return common.forValue(this.blocks, hash, hash, timeout);
+  }
+}
+
+describe('Wallet RPCs', function() {
+  this.timeout(60000);
+
+  describe('Unsigned Auction TXs', function() {
+    const util = new TestUtil();
+    const name = rules.grindName(2, 0, Network.get('regtest'));
+    let winner, loser;
+
+    const mineBlocks = async (num, wallet, account = 'default') => {
+      const address = (await wallet.createAddress(account)).address;
+      const hashes = await util.nrpc('generatetoaddress', [num, address]);
+      await util.confirmBlock(hashes.pop());
+    };
+
+    const processJSON = async (json, submit, wallet = winner, show) => {
+      const mtx = MTX.fromJSON(json);
+
+      for (let i = 0; i < mtx.inputs.length; i++) {
+        const input = mtx.inputs[i];
+        const coin = mtx.view.getCoinFor(input);
+        const path = mtx.view.getPathFor(input);
+        const address = coin.address.toString('regtest');
+        const key = await wallet.getKey(address);
+
+        // Assert HD path.
+        assert.ok(path instanceof Path);
+        assert.deepStrictEqual(path.name, key.name);
+        assert.deepStrictEqual(path.account, key.account);
+        assert.deepStrictEqual(path.branch, key.branch);
+        assert.deepStrictEqual(path.index, key.index);
+
+        // Sign mtx.
+        const secret = (await wallet.getWIF(address)).privateKey;
+        const ring = KeyRing.fromSecret(secret);
+        mtx.sign(ring);
+      };
+
+      // Verify mtx.
+      assert(mtx.verify());
+
+      // Submit and mine mtx, if necessary.
+      if (submit) {
+        await util.nrpc('sendrawtransaction', [mtx.encode().toString('hex')]);
+        await mineBlocks(1, wallet);
+        await util.confirmTX(mtx.txid());
+      }
+
+      return mtx;
+    };
+
+    before(async () => {
+      util.node.network.coinbaseMaturity = 1;
+      await util.open();
+      winner = await util.wclient.wallet('primary');
+      loser = await util.wclient.createWallet('loser');
+      loser = await util.wclient.wallet('loser');
+      await mineBlocks(5, winner);
+      await mineBlocks(5, loser);
+    });
+
+    after(async () => {
+      util.node.network.coinbaseMaturity = 2;
+      await util.close();
+    });
+
+    it('should create OPEN with signing paths', async () => {
+      // Create, assert, submit and mine OPEN.
+      const submit = true;
+      const json = await util.wrpc('createopen', [name]);
+      await processJSON(json, submit, winner);
+
+      // Mine past OPEN period.
+      await mineBlocks(7, winner);
```

Comment:
The bidding period starts in the next `treeInterval`, its probably better to use that variable rather than a magic number